base/dates/io.jl

```diff
@@ -431,7 +431,7 @@ Parse a date from a date string `dt` using a `DateFormat` object `df`.
 """
 Date(dt::AbstractString,df::DateFormat=ISODateFormat) = parse(Date, dt, df)
 
-@generated function format{S, T}(io::IO, dt::TimeType, fmt::DateFormat{S, T})
+@generated function format(io::IO, dt::TimeType, fmt::DateFormat{S, T}) where {S, T}
```

Comment:
Unless I missed use of `S` in the method body,  perhaps could also simplify the signature to `format(io::IO, dt::TimeType, fmt::DateFormat{<:Any,T}) where T`?